controllers/reports.js

```diff
@@ -0,0 +1,142 @@
+import models from '../models';
+
+const { User, Report, article } = models;
+
+const reporting = {
+  /**
+   * Authenticated user should be able to report
+   * an article that violate terms of aggrement.
+   *
+   * @param {*} req request object.
+   * @param {*} res response object.
+   * @returns {*}   response to be returned.
+   */
+  reportArticle: async (req, res) => {
+    const { slug } = req.params;
+    const { message } = req.body;
+    const { id: reporter } = req.user;
+    try {
+      const findArticle = await article.findOne({
+        where: { slug },
+        attributes: ['id', 'slug', 'title', 'body']
+      });
+      if (!findArticle) {
+        return res.status(404).send({
+          status: res.statusCode,
+          error: 'This article is not found',
+        });
+      }
+
+      const findReport = await Report.findOne({ where: { reporter, articleId: findArticle.id } });
+      if (findReport) {
+        return res.status(400).send({
+          status: res.statusCode,
+          error: 'You have already reported this article',
```

Comment:
@niyobobo could it be best to return the name of the reported in the error message: For example: "Aaron Sekisambu, you have already reported this article"